backend/api/users.py

```diff
@@ -185,20 +185,98 @@ async def delete_user(user: models.User = Depends(get_current_user)):
                             detail="User not found")
 
 
-@router.get("/username/{username}", status_code=status.HTTP_200_OK,
-            response_model=models.UserSummary)
-def get_user_by_username(username: str):
+@router.post("/get_by_usernames", status_code=status.HTTP_200_OK,
+             response_model=list[models.UserSummary])
+def get_by_usernames(usernames: list[str]):
```

Comment:
why don't you use the single one many times?

Reply:
removed the function